### ISSUE TITLE
Data for Ryza and Sophie 2. And potential traits pathfinder widget.

### ISSUE THREAD

**Issue by DableUTeeF** [CatClawed/barrelwisdom#111]
I'm currently planning to make an app to help me figuring out how to get all traits I want from the materials to the final item. Maybe something similar to this https://aqiu384.github.io/megaten-fusion-tool/smt5v/recipes (select a bunch of items with traits you want and print out possible steps to combine them).

But at the moment the only raw data I can find is for Sophie 1 and Ryza 3. Would you be willing to share the data you use or at least point me to how to get it?

Also are you interested in having such a thing in your site? If so then I'd make it from a fork of this repo.

I just finished Ryza trilogy and Sophie 2 thinking about replaying all of them after I make this thing.

**Comment by holly-hacker** [comment]
You can extract the item data from the game using [gust_tools](https://github.com/VitaSmith/gust_tools) or [atelier-tools](https://github.com/holly-hacker/atelier-tools). You will then have to parse that XML data to extract what you need, of which I have an example implementation [here](https://github.com/holly-hacker/atelier-data-extractor).

Is the tool you are looking for similar to this: https://atelier.variant9.dev/#/sophie/tools/synth-graph
If so, I can open source the code behind it so you can check it out.

**Comment by DableUTeeF** [comment]
Yes something like that. I'm also planning to have multiple materials so I can put 3 traits on a single item.

**Comment by holly-hacker** [comment]
You can check out the code behind that synth-graph page [here](https://github.com/holly-hacker/atelier-wiki/blob/main/frontend/src/routes/sophie/tools/synth-graph.tsx), but it's just a simple graph traversal so there's not that much to learn I think.
